### PR TITLE
SPCTR-3006: Fix: Buttons-child icon spacing for rtl.

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description

Issue: When we use the button block with the icon on the RTL site, there is no space between icon and text.
https://brainstormforce.atlassian.net/browse/SPCTR-3006

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-block-helper.php [hunk 1]
```diff
@@ -276,25 +278,48 @@ public static function get_buttons_child_selectors( $attr, $id, $child_migrate )
...
-				$selectors[ $wrapper . ' .uagb-button__icon-position-after' ]   = array(
-					'margin-left' => UAGB_Helper::get_css_value( self::get_fallback_number( $attr['iconSpace'], 'iconSpace', $block_name ), 'px' ),
-				);
-				$t_selectors[ $wrapper . ' .uagb-button__icon-position-after' ] = array(
-					'margin-left' => UAGB_Helper::get_css_value( $attr['iconSpaceTablet'], 'px' ),
-				);
-				$m_selectors[ $wrapper . ' .uagb-button__icon-position-after' ] = array(
-					'margin-left' => UAGB_Helper::get_css_value( $attr['iconSpaceMobile'], 'px' ),
-				);
+				if ( ! $is_rtl ) {
```
Comment (AbhijitNage123) [inline classes/class-uagb-block-helper.php:281]: When site is not RTL then we are applying margin left/right iconSpace CSS. What if user has RTL site, and he change the left/right IconSpace from backend. Could you please check above case? @rajmohanreddy774

**Review by aapatre — APPROVED** [review]

File: classes/class-uagb-block-helper.php (hunk 1 above)
Comment (rajmohanreddy774) [reply classes/class-uagb-block-helper.php:281]: @AbhijitNage123 Both frontend and backend is handled for rtl and without rtl for iconSpacing.

**Review by AbhijitNage123 — APPROVED** [review]

File: src/blocks/buttons-child/styling.js [hunk 2]
```diff
@@ -330,27 +332,46 @@ function styling( attributes, clientId, name ) {
...
-			'margin-right': generateCSSUnit( iconSpaceMobile, 'px' ),
-		};
-		mobileSelectors[ ' .uagb-button__icon-position-after' ] = {
-			'margin-left': generateCSSUnit( iconSpaceMobile, 'px' ),
-		};
-
-		selectors[ ' .uagb-button__icon-position-before' ] = {
-			'margin-right': generateCSSUnit( getFallbackNumber( iconSpace, 'iconSpace', blockName ), 'px' ),
-		};
+		if( ! isRTL ) {
```
Comment (adi3890) [inline src/blocks/buttons-child/styling.js:335]: can we add condition here it self. if it is one time use.
can we remove redundancy of code below.

File: classes/class-uagb-block-helper.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply classes/class-uagb-block-helper.php]: Done changes.

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -276,25 +278,48 @@ public static function get_buttons_child_selectors( $attr, $id, $child_migrate )
...
+					$selectors[ $wrapper . ' .uagb-button__icon-position-after' ]   = array(
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply classes/class-uagb-block-helper.php]: Done changes as suggested.

File: src/blocks/buttons-child/styling.js (hunk 2 above)
Comment (rajmohanreddy774) [reply src/blocks/buttons-child/styling.js]: Done changes.

File: src/blocks/buttons-child/styling.js [hunk 4]
```diff
@@ -329,29 +334,35 @@ function styling( attributes, clientId, name ) {
 	] = {
 		'fill': iconHColor,
 	};
+
 	if ( ! removeText ) {
+		const iconMargin = generateCSSUnit( getFallbackNumber( iconSpace, 'iconSpace', blockName ), 'px' );
+		const tabletIconMargin = generateCSSUnit( iconSpaceTablet, 'px' );
+		const mobileIconMargin = generateCSSUnit( iconSpaceMobile, 'px' );
+
+		const rtlMargin = ! isRTL ? 'margin-left' : 'margin-right';
```
Comment (adi3890) [inline src/blocks/buttons-child/styling.js:343]: lets name variable as leftMargin, rightMargin

File: classes/class-uagb-block-helper.php [hunk 5]
```diff
@@ -276,26 +276,34 @@ public static function get_buttons_child_selectors( $attr, $id, $child_migrate )
 				'fill' => $attr['iconHColor'],
 			);
 			if ( ! $attr['removeText'] ) {
+				$iconMargin       = UAGB_Helper::get_css_value( self::get_fallback_number( $attr['iconSpace'], 'iconSpace', $block_name ), 'px' );
+				$tabletIconMargin = UAGB_Helper::get_css_value( $attr['iconSpaceTablet'], 'px' );
+				$mobileIconMargin = UAGB_Helper::get_css_value( $attr['iconSpaceMobile'], 'px' );
+			  
+				$rtlMargin = ! is_rtl() ? 'margin-left' : 'margin-right';
```
Comment (adi3890) [inline classes/class-uagb-block-helper.php:283]: use variable name as $left_margin, $right_margin
assign default value.
and change when is_rtl()

**Review by adi3890 — DISMISSED** [review]

File: package.json [hunk 6]
```diff
@@ -80,7 +80,7 @@
     "lint-css": "wp-scripts lint-style",
     "lint-css:fix": "npm run lint-css -- --fix",
     "pretty": "prettier --check .",
-    "pretty:fix": "prettier --write .",
+    "pretty:fix": "prettier --write ./src/blocks/buttons-child/styling.js",
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply package.json]: Removed.

File: src/blocks/buttons-child/styling.js (hunk 4 above)
Comment (rajmohanreddy774) [reply src/blocks/buttons-child/styling.js]: Done.

File: classes/class-uagb-block-helper.php (hunk 5 above)
Comment (rajmohanreddy774) [reply classes/class-uagb-block-helper.php]: Done.

File: src/blocks/buttons-child/styling.js [hunk 7]
```diff
@@ -333,28 +335,32 @@ function styling( attributes, clientId, name ) {
...
-		selectors[ ' .uagb-button__icon-position-before' ] = {
-			'margin-right': generateCSSUnit( getFallbackNumber( iconSpace, 'iconSpace', blockName ), 'px' ),
-		};
-	}
+        const iconMargin = generateCSSUnit( getFallbackNumber( iconSpace, 'iconSpace', blockName ), 'px' );
+        const tabletIconMargin = generateCSSUnit( iconSpaceTablet, 'px' );
+        const mobileIconMargin = generateCSSUnit( iconSpaceMobile, 'px' );
+        let rightSideMargin = 'margin-right';
+        let leftSideMargin = 'margin-left';
+        rightSideMargin = ! isRTL ? 'margin-left' : 'margin-right';
```
Comment (adi3890) [inline src/blocks/buttons-child/styling.js:343]: should we try this logic
if( '1' === uagb_blocks_info.is_rtl ) {
 rightSideMargin = 'margin-left';
 leftSideMargin = 'margin-right';
}

This way we don't neeed to declare a const isRTL

File: classes/class-uagb-block-helper.php [hunk 8]
```diff
@@ -276,26 +276,37 @@ public static function get_buttons_child_selectors( $attr, $id, $child_migrate )
 				'fill' => $attr['iconHColor'],
 			);
 			if ( ! $attr['removeText'] ) {
+				$iconMargin       = UAGB_Helper::get_css_value( self::get_fallback_number( $attr['iconSpace'], 'iconSpace', $block_name ), 'px' );
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply classes/class-uagb-block-helper.php]: Done changes.

File: src/blocks/buttons-child/styling.js (hunk 7 above)
Comment (rajmohanreddy774) [reply src/blocks/buttons-child/styling.js]: Done changes suggested.

File: classes/class-uagb-block-helper.php [hunk 9]
```diff
@@ -276,26 +276,40 @@ public static function get_buttons_child_selectors( $attr, $id, $child_migrate )
...
+				$tablet_icon_margin = UAGB_Helper::get_css_value( $attr['iconSpaceTablet'], 'px' );
+				$mobile_icon_margin = UAGB_Helper::get_css_value( $attr['iconSpaceMobile'], 'px' );
+
+				$right_side_margin = 'margin-right';
+				$left_side_margin  = 'margin-left';
+
+				if ( ! is_rtl() ) {
+					$right_side_margin = 'margin-left';
+					$left_side_margin  = 'margin-right';
+
```
Comment (adi3890) [inline classes/class-uagb-block-helper.php:289]: please remove this extra space

**Review by adi3890 — APPROVED** [review]

**Review by bsf-zanev — APPROVED** [review]

File: classes/class-uagb-block-helper.php (hunk 9 above)
Comment (rajmohanreddy774) [reply classes/class-uagb-block-helper.php]: done

**Pull request merged**